packages/grid/x-data-grid/src/colDef/gridSingleSelectOperators.ts

```diff
@@ -11,6 +13,51 @@ const parseObjectValue = (value: GridFilterItem) => {
   return value.value;
 };
 
+export const getGridSingleSelectQuickFilterFn = (
+  value: any,
+  column: GridColDef,
+  apiRef: React.MutableRefObject<GridApiCommunity>,
+) => {
+  if (!value) {
+    return null;
+  }
+  const { valueOptions, valueFormatter, field } = column;
+
+  const potentialValues = [parseObjectValue(value).toString()];
+
+  const iterableColumnValues =
+    typeof valueOptions === 'function' ? valueOptions({ field }) : valueOptions || [];
+
+  if (iterableColumnValues) {
+    iterableColumnValues.forEach((option) => {
+      // for each valueOption, check if the formatted value
+      let optionValue;
+      let optionLabel;
+      if (typeof option === 'object') {
+        optionValue = option.value;
+        optionLabel = option.label;
+      } else {
+        optionValue = option;
+        if (valueFormatter) {
+          optionLabel = valueFormatter({ value: option, field, api: apiRef.current });
+        } else {
+          optionLabel = option;
+        }
+      }
+
+      if (optionLabel.slice(0, value.length).toLowerCase() === value.toLowerCase()) {
+        if (!potentialValues.includes(optionValue)) {
+          potentialValues.push(optionValue.toString());
+        }
+      }
+    });
+  }
+
+  return ({ value: columnValue }: GridCellParams): boolean => {
+    return columnValue != null ? potentialValues.includes(columnValue.toString()) : false;
```

Comment:
Should the `valueFormatter` be applied to `columnValue`?

In https://deploy-preview-4317--material-ui-x.netlify.app/components/data-grid/filtering/#quick-filter I can't search for country names because the value tested is an object.

Reply:
My strategy is to first loop on `valueOptions` to find which option values are accepted. Such that for each row I only check if the value is in "valid" options

I forget to use `parseObjectValue` in the second step to extract `value` property when the cell value is an object. Should be fixed now